[doc] Add CUDA description to package

hickeyma opened the pull request:
Fixes #114364

hickeyma committed:
Add CUDA description

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

malfet commented on torch/cuda/__init__.py:
```diff
@@ -1,6 +1,12 @@
+developed by NVIDIA for general computing on GPUs.
+With CUDA, developers are able to dramatically speed up computing
+applications by harnessing the power of GPUs.
+See https://developer.nvidia.com/cuda-zone for more details.
```
Would https://en.wikipedia.org/wiki/CUDA be a better link here?

hickeyma replied:
I think so, provides more details and context.

hickeyma replied:
Added

malfet reviewed (commented):
To be frank I'm a bit on the edge here: on one hand it feels good to add a bit more explanation, but on the other how to define when one should stop?
Should we define what GPU is? Or thread? Or half precision data type?

hickeyma committed:
Update after review

Review comments:
https://github.com/pytorch/pytorch/pull/114365#pullrequestreview-1744934978

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

hickeyma commented on torch/cuda/__init__.py:
```diff
@@ -1,6 +1,12 @@
 
+CUDA is a parallel computing platform and programming model
+developed by NVIDIA for general computing on GPUs.
+With CUDA, developers are able to dramatically speed up computing
+applications by harnessing the power of GPUs.
```
Agree, removed.

hickeyma commented:
> To be frank I'm a bit on the edge here: on one hand it feels good to add a bit more explanation, but on the other how to define when one should stop?

@malfet That's a reasonable point. Adding some small context and link to details on wikipedia seems like a good balance.

hickeyma commented:
Thanks for the review @malfet. Comments updated and ready for review again.

hickeyma commented:
Ping @malfet @svekars @carljparker
